src/fluxpy/utils/model.py

```diff
@@ -415,3 +419,309 @@ def sync_with_medium(model: cobra.Model, medium: Dict):
+                    
+                        # boolean to check if pairwise reactants/products are the same
+                        identical_reactants = (set(self.reactants_list_all_reactions[i]) == set(self.reactants_list_all_reactions[j]))
+                        identical_products = (set(self.products_list_all_reactions[i]) == set(self.products_list_all_reactions[j]))
+                        identical_cofactors = (set(self.cofactors_list_all_reactions[i]) == set(self.cofactors_list_all_reactions[j]))
+
+                        if identical_reactants == True and identical_products == True and identical_cofactors == False:
+                            # finish here and keep combination
+                            combinations.append((self.reactions_ids[i], self.reactions_ids[j]))
+                        else:
+                            if self.reversibility_list_all_reactions[i] == True or self.reversibility_list_all_reactions[j] == True:
+                                # switch reaction i and compare updates reactants-products
+                                identical_reactants = (set(self.reactants_list_all_reactions[i]) == set(self.products_list_all_reactions[j]))
+                                identical_products = (set(self.products_list_all_reactions[i]) == set(self.reactants_list_all_reactions[j]))
+
+                                if identical_reactants == True and identical_products == True and identical_cofactors == False:
+                                    # finish here and keep combination
+                                    combinations.append((self.reactions_ids[i], self.reactions_ids[j]))
+
+        self.unique_combinations = {tuple(sorted(pair)) for pair in combinations}
+
+
+    def merge_connected_combinations(self):
+        
+        self.find_reactions_combinations()
+        
+        # Step 1: Create a graph using an adjacency list representation
+        # We'll use a defaultdict where each key is a node, and its value is a set of neighboring nodes
+        graph = defaultdict(set)
+
+        # Build the adjacency list by iterating through each pair in the input
+        for x, y in self.unique_combinations:
+            # Add y as a neighbor of x
+            graph[x].add(y)
+            # Add x as a neighbor of y (because the graph is undirected)
+            graph[y].add(x)
+
+        # At this point, for an input of this format: [('A', 'B'), ('A', 'C'), ('B', 'C'), ('D', 'K')]
+        # the `graph` looks like:
+        # {'A': {'B', 'C'}, 'B': {'A', 'C'}, 'C': {'A', 'B'}, 'D': {'K'}, 'K': {'D'}}
+
+        # Step 2: Define a function to find connected components using Depth-First Search (DFS)
+        def find_connected_components(graph):
+            # A set to keep track of visited nodes
+            visited = set()
+            # A list to store the connected components
+            components = []
+
+            # Helper function to perform a recursive DFS
```

Comment:
As before, consider moving helper function out of the class for clarity and re-usability. The `utils.py` script is the place to be.